GUIDELINES.md

```diff
@@ -0,0 +1,135 @@
+* Start with the test
+  * In the test class you created add reference to the steps class with the `steps` function (e.g. `val sut by steps<LoginSteps>`) and call the value `sut` (system under test)
+  * Fill up the test with `@Test`-annotated test functions
+    * The test function can be named after a _scenario_ (e.g. `New user trying to log in`, `Weather condition is cloudy`)
+    * Only if necessary to distinguish test cases, also add more context like what's roughly expected (e.g. `New user trying to login - welcome to the app message shown`, `Existing user trying to login - welcome back message shown`)
+    * Don't use names like `givenNewUser_whenLoggingIn_thenShowWelcomeToAppMessage` because
+      * The test code in the test functions should already be easy enough to understand preconditions, actions and assumptions
+      * The more text you use the harder to distinguish test cases
+      * The less spaces you use the harder to read
+      * So please just use the test function names to make a meaningful distinctions to properly navigate between and understand test scenarios and cases
+  * For each test case, call functions that would reside in the steps class and prefix these functions with `given`, `when` or `then`
+    * Functions starting with `given` should set up the environment, as these define the assumptions that are made (e.g. `givenUser(email = "test@test.com", password = "supersecure")` indicating that the fake backend should act as if there was that user saved in the database). For the naming choose present tense (e.g. `givenUser` as a short form of "there is a user" or `givenInternetConnected`)
+    * Functions starting with `when` should perform actions, therefore they are named in progressive form (e.g. `whenGoingOffline`, `whenInitializing`, `whenLoggingInWith(email, password)`)
+    * Functions starting with `then` should perform assertions (e.g. `thenIsLoggedIn()`, `thenIsBusy()` or `thenUserIsCreated(user: User)`)
+    * By starting at the test to define which functions are needed in the steps you are more focused and get a faster overview of that's needed instead of getting into details too early
+  * Now the test should already be human-readable for non-tech people (with just the technical implementation in the steps class missing)
+  * Remember that in the test class all data structures and function calls should be domain-specific and therefore be modelled around real-world domain objects and data visible to a user, as far as possible (if the test subject is of total technical nature this doesn't apply of course)
+* Now think about which dependencies (classes or interfaces) will be under test and add new steps classes
+  * You can add a steps classs per class under test (e.g. `AuthManagerSteps`)
+  * As stated above (just to repeat)
+    * the steps classes' job is to know "how this class is to be tested"
```

Comment:
I would say that a Steps class is basically the API to the production code. It should help to update the methods, signatures etc in production code without the need to update all the unit tests. This is often a complaint by people who don't write proper unit tests, and Steps classes help here :)